lib/request.js

```diff
@@ -252,6 +252,7 @@ module.exports = {
   get host() {
     const proxy = this.app.proxy;
     let host = proxy && this.get('X-Forwarded-Host');
+    if (this.req.httpVersionMajor >= 2) host = this.get(':authority');
```

Comment:
https://github.com/nodejs/node/blob/e7f710c1d4729890806610e5450d7a0edfb4ac11/lib/internal/http2/compat.js#L358

use `this.req. authority` will be better.

Reply:
This seems to be an undocumented property. Is it save to use it and should koa use undocumented props?

Reply:
It's unsafe either way, at least `this.req.authority` is a specific getter with a sane naming that conforms with the RFC headerfield.

Reply:
`this.req.headers` is the documented way to access http headers and `this.get` is using this way. And `:authority` is the official header name from the [RFC 7540](https://tools.ietf.org/html/rfc7540#section-8.1.2.3). So this way should not be unsafe.

Whereas `this.req.authority` is not documented and therefore could change in a minor update and is considered an internal API (as described in the [node COLLABORATOR_GUIDE](https://github.com/nodejs/node/blob/master/COLLABORATOR_GUIDE.md#internal-vs-public-api)).

But I agree that `this.req.authority` should be part of the public API and I just created an Issue in the node repo (nodejs/node#23825).